### PR TITLE
Add `initialPublicData` option to `useSession` for use with SSR (minor)

### PR DESCRIPTION
**Pull request by engelkes-finstreet**
Adds the functionality suggested in [this pr](https://github.com/blitz-js/blitz/pull/1283).

Now you can pass ` useSession({publicData})` to the hook and the _isLoading_ state is directly set to false to avoid any flickering.

### CODE REVIEW

File: packages/core/src/supertokens.ts [hunk 1]
```diff
@@ -65,9 +65,17 @@ export interface PublicDataWithLoading extends PublicData {
...
-  const [isLoading, setIsLoading] = useState(true)
+interface PublicDataWithInitialValues {
+  publicData: PublicData
+}
+
+export const useSession: (initial?: PublicDataWithInitialValues) => PublicDataWithLoading = (
+  initial?: PublicDataWithInitialValues,
+) => {
+  const [publicData, setPublicData] = useState(
+    initial?.publicData ?? publicDataStore.emptyPublicData,
```
Comment (flybayer) [inline packages/core/src/supertokens.ts:76]: Sweet, thanks @engelkes-finstreet! 

Let's rename `initial.publicData` to `options.initialPublicData`

And then go ahead and open a docs PR!

File: packages/core/src/supertokens.ts [hunk 2]
```diff
@@ -65,9 +65,17 @@ export interface PublicDataWithLoading extends PublicData {
...
+  )
+  const [isLoading, setIsLoading] = useState(!initial)
```
Comment (flybayer) [inline packages/core/src/supertokens.ts:78]:
```suggestion
  const [isLoading, setIsLoading] = useState(!options?.initialPublicData)
```

**Comment by flybayer** [comment]
Btw, you could have marked this as "ready for review" because that's exactly what it was :)

**Comment by flybayer** [comment]
@engelkes-finstreet are you able to finish this up?

**Comment by engelkes-finstreet** [comment]
> @engelkes-finstreet are you able to finish this up?

Yes, sorry! I had a busy work week. Updated with your suggested changes

**Comment by flybayer** [comment]
Ok, thanks, last thing is to open a PR to update the docs!

**Comment by engelkes-finstreet** [comment]
[PR for docs](https://github.com/blitz-js/blitzjs.com/pull/356)

File: packages/core/src/supertokens.ts [hunk 3]
```diff
@@ -61,9 +61,17 @@ export interface PublicDataWithLoading extends PublicData {
   isLoading: boolean
 }
 
-export const useSession: () => PublicDataWithLoading = () => {
-  const [publicData, setPublicData] = useState(publicDataStore.emptyPublicData)
-  const [isLoading, setIsLoading] = useState(true)
+interface PublicDataWithInitialValues {
```
Comment (flybayer) [inline packages/core/src/supertokens.ts:64]:
```suggestion
interface UseSessionOptions {
```

**Review by flybayer — APPROVED** [review]

**Pull request merged**